qcodes/instrument_drivers/Minicircuits/rcdat_6000_60_usb.py

```diff
@@ -0,0 +1,160 @@
+                                  'get_firmware': bytes([99]), }
+
+    # size of usb package in bytes
+    package_size = 64
+
+    def __init__(
+            self,
+            name: str,
+            instance_id: Optional[str] = None,
+            **kwargs) -> None:
+        """
+        Instantiates the instrument.
+
+        Args:
+            name: The instrument name used by qcodes
+            instanc_id: The identification string for the instrument. If there
+               are multiple instruments of the same type connected, an
+               additional 'instance_id' has to be provided to identify the
+               desired instance. Use the enumerate_devices() function to get a
+               list of all 'instance_id's of the connected devices.
+        """
+
+        # super refers to mixin, mixin calls super().__init__() to
+        # refer to Instrument
+        super().__init__(instance_id=instance_id, name=name, **kwargs)
+
+        self.connect_message()
+
+        self.add_parameter(name='attenuation',
+                           label='attenuation',
+                           unit='dB',
+                           get_cmd=self.get_attenuation,
+                           set_cmd=self.set_attenuation,)
+        # TODO: add validator
+
+    @staticmethod
+    def _prepare_package(cmd: str, data: bytes = None) -> bytearray:
+        bCmd = RCDAT_6000_60_usb.commands[cmd]
+        if data:
+            package = bCmd + data
+        else:
+            package = bCmd
+        return package
+
+    def write_cmd(self, cmd: str, data: bytes = None) -> None:
+        self.write_USB_data(self.feature_id, self._prepare_package(cmd, data))
+
+    def ask_cmd(self, cmd: str, data: bytes = None) -> bytearray:
+        # clip the first byte off because it reflects the command
+        ret = self.ask_USB_data(
+            self.feature_id,
+            self._prepare_package(
+                cmd,
+                data))
+        return ret[1:]
+
+    def get_attenuation(self) -> float:
+        ret = self.ask_cmd('get_attenuation')
+        return ret[2] + float(ret[3]) / 4
+
+    def set_attenuation(self, attenuation: float) -> None:
+        data = bytearray(2)
+        data[0] = int(attenuation)
+        data[1] = int((attenuation - data[0]) * 4)
+        self.write_cmd('set_attenuation', data)
+
+    def get_idn(self) -> Dict[str, str]:
+        """
+        overides get_idn from 'Instrument'
+        """
+        model = self._bytearray_to_string(
+            self.ask_cmd('get_device_model_name'))
+        serial = self._bytearray_to_string(
+            self.ask_cmd('get_device_serial_number'))
+        firmware = self._bytearray_to_string(self.ask_cmd('get_firmware'))
+        return {'vendor': 'Mini-Circuits',
+                'model': model,
+                'serial': serial,
+                'firmware': firmware}
+
+    # this does not work for this instrument for no apparent reason
```

Comment:
I find this docstring confusing

Reply:
In which way do you think it is confusing? Do you have a suggestion?

Reply:
Two cases of "this" in a sentence is almost always confusing. And "does not work" can mean a number of things. I am guessing that the following would be more descriptive:

```
# NOTE: the following command silently failed when tested on unit XXXXX
# according to the manual, it should work.
```

Reply:
👍